storage/2020-08-04/blob/blobs/copy_and_wait_poller.go

```diff
@@ -0,0 +1,48 @@
+package blobs
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-azure-sdk/sdk/client/pollers"
+)
+
+var _ pollers.PollerType = &copyAndWaitPoller{}
+
+func NewCopyAndWaitPoller(client *Client, containerName, blobName string, getPropertiesInput GetPropertiesInput) *copyAndWaitPoller {
+	return &copyAndWaitPoller{
+		client:             client,
+		containerName:      containerName,
+		blobName:           blobName,
+		getPropertiesInput: getPropertiesInput,
+	}
+}
+
+type copyAndWaitPoller struct {
+	client             *Client
+	containerName      string
+	blobName           string
+	getPropertiesInput GetPropertiesInput
+}
+
+func (p *copyAndWaitPoller) Poll(ctx context.Context) (*pollers.PollResult, error) {
+	props, err := p.client.GetProperties(ctx, p.containerName, p.blobName, p.getPropertiesInput)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving properties (container: %s blob: %s) : %+v", p.containerName, p.blobName, err)
+	}
+
+	if strings.EqualFold(string(props.CopyStatus), "success") {
```

Comment:
could we use the constant here:

```suggestion
	if strings.EqualFold(string(props.CopyStatus), string(Success)) {
```